### PR TITLE
[FEATURE] Special metadata in notifications

### PR DESCRIPTION
**Pull request by iariasleon**
```
                    SUMMARY:
 ----------------------------------------------------------------------------------------------------------------------------------
|                                   name                                       | passed | failed | skipped |  total  | duration(s) |
 ----------------------------------------------------------------------------------------------------------------------------------
| components\ngsiv2\subscriptions\notifications\notifications_metadata.feature |     31 |      0 |      18 |      49 |      50.137 |
 ----------------------------------------------------------------------------------------------------------------------------------
|  Date: Thursday, September 29, 2016 15:56:46
 -------------------------------------------------
1 feature passed, 0 failed, 0 skipped
31 scenarios passed, 0 failed, 18 skipped
351 steps passed, 0 failed, 222 skipped, 0 undefined
Took 0m50.137s
```
@fgalan @kzangeli @crbrox

### CODE REVIEW

File: test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+      | attributes_value  | random=5    |
+      | attributes_type   | celsius     |
+      | metadatas_number  | 2           |
+      | metadatas_name    | very_hot    |
+      | metadatas_type    | alarm       |
+      | metadatas_value   | hot         |
+    When create entity group with "1" entities in "normalized" mode
+    And verify that receive several "Created" http code
+    Then get notification sent to listener
+    And verify metadata in notification with "*"
```
Comment (fgalan) [inline test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature]: Could you elaborato on what this means?

Comment (fgalan) [inline test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature]: In particular, what the meaning of "*" is (when `<metadata>` is used, I think I have got the point...)

File: test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature [hunk 2]
```diff
@@ -0,0 +1,493 @@
...
+      | attributes_type   | celsius     |
+      | metadatas_number  | 2           |
+      | metadatas_name    | very_hot    |
+      | metadatas_type    | alarm       |
+      | metadatas_value   | hot         |
+    When create entity group with "1" entities in "normalized" mode
+    And verify that receive several "Created" http code
+    Then get notification sent to listener
+    And verify metadata in notification with "<metadata>"
+    Examples:
```
Comment (fgalan) [inline test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature]: As far as I understand the text check that the notification has the right metadata... However, where do you test that these metadata has the right value? e.g. actionType is "append" and so on.

File: test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature (hunk 1 above)
Comment (iariasleon) [inline test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature]: In this case , It is not used special metadata, ,so only it is returned the custom user metadata, I apply this concept, but only in the test step, not in the subscription. I change this step to clarify.

File: test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature (hunk 2 above)
Comment (iariasleon) [inline test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature]: I take the `previousValue` and the `actionType` from previous steps (create and update request), and all tests are implemented in the notification_steps.py and the NGSI_v2.py libraries. I push these libraries in another PR. And another library in commom repository to create requests [(cb_v2_utils.py).](https://github.com/telefonicaid/iotqatools/blob/develop/iotqatools/cb_v2_utils.py)

File: test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature (hunk 1 above)
Comment (iariasleon) [inline test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature]: Fixed in ff68c15

File: test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature (hunk 2 above)
Comment (fgalan) [inline test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature]: So you mean that the verification implemented in Python for the "verify metadata in notification with ..." is dependand on the metadata name itself?

Comment (iariasleon) [inline test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature]: yes, the step verify depending of the special metadata defined. If a metadata is included in the list, It is verified.

Comment (fgalan) [inline test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature]: A bit obscure, as reading the .feature file it is not clear in which sub-steps the verification consits. However, I'm not the QA guy so if it ok for you it is ok with me :)

I guess NTC

Comment (iariasleon) [inline test/acceptance/behave/components/ngsiv2/subscriptions/notifications/notifications_metadata.feature]: I can to create a step by each verification (I sure will be more clear), but we will have many, many steps in each scenario. In this case I have 10 assertions in this step `verify metadata in notification with "<metadata>".

**Comment by fgalan** [comment]
LGTM

**Pull request merged**